test/test_autograd.py

```diff
@@ -3921,6 +3921,14 @@ def test_inputbuffer_add_multidevice(self, devices):
         output = input.to(device=devices[1]) + input.to(device=devices[1])
         output.backward()
 
+    def test_copy__bfloat16(self, device):
+        x = torch.randn(10, device=device, requires_grad=True)
+        y = torch.empty(10, device=device, dtype=torch.bfloat16)
+        y.copy_(x)
+        self.assertTrue(y.requires_grad)
+        z = x.to(torch.bfloat16)
+        self.assertTrue(z.requires_grad)
```

Comment:
It would be nice if we could have a list of `dtypes` here that includes all floating-point types, to make sure that this property is preserved for all of them, not just `bfloat16`